Refactoring Branch

AndreG-P opened the pull request:
@physikerwelt @vstange 
I started to refactor MathMLTools as mentioned in #13

This branch contains now:
- MathMLTools (in mathml-core now)
- MathMLConverters (in mathml-converters now)
- MathMLSim (in mathml-similarity now)

All tests except one (see #16) are finished successful on my machine (we will see what travis will show us...).

I don't know exactly why... Just before I start to refactor the project, I merged the WIP branch from Moritz https://github.com/ag-gipp/MathMLTools/pull/8 to master. I made some fixes but it looked like it worked. I merged it because I wanted the latest version in the refactored project (I presumed to solve merge conflicts after the refactoring will be a nightmare).

Besides this, I will change the naming a little bit further.

However, I'm more happy with the new structure. I need to setup the auto deployment system again, since it is a multi-module project now (see #15). Any suggestions for further improvements would be nice.

PS: I needed to merge the travis config also. We should definitely find another solution for this travis hack with mathoid. See https://github.com/ag-gipp/mathosphere/issues/108

physikerwelt committed:
Avoid warnings and fix input before parsing

physikerwelt committed:
Add functionality to change tex annotations

physikerwelt committed:
Fixup Ensure that method is not misused

physikerwelt committed:
Reduce output of checkstyle

Use apache commons lang3 for NotImplemented exceptions

physikerwelt committed:
Build csymbol abstraction

physikerwelt committed:
Fix WS issues

physikerwelt committed:
Rename Math to MathDoc

Avoid confusion with java Math class

physikerwelt committed:
WIP move XMLDocument reader to tools

AndreG-P committed:
Merge with WIP track to be up to date before refactoring

AndreG-P committed:
Huge refactoring commit

AndreG-P committed:
Travis bug fix for MathMLConverters

physikerwelt commented:
@AndreG-P you spend soo much time to write the logs. Now you also have to read them
https://travis-ci.org/ag-gipp/MathMLTools/builds/344730018#L2477
;-)

vstange commented:
mathoid didn't start see travis output on:
`Error while reading config file: Error: ENOENT: no such file or directory, open '/home/travis/build/ag-gipp/MathMLTools/config.dev.yaml'`

MathMLConverters has a custom config for mathoid: https://github.com/ag-gipp/MathMLConverters/blob/master/config.dev.yaml

AndreG-P committed:
Try to fix travis

AndreG-P committed:
Avoid potential problems with maven tests
by adding junit-jupiter-engine to maven surefire.

AndreG-P committed:
Move config.dev.yaml to .travis and change expected results of mathoid
The config for mathoid is only necessary because of travis installations. So I moved it to the .travis folder.
Also, an older version of mathoid is used (from https://git@github.com/physikerwelt/mathoid-server).
We should fix this kind of tests!

AndreG-P committed:
Rename package structure of MathMLSim

AndreG-P committed:
Add codeclimate to readme and change icons to branch refactor

AndreG-P committed:
Refactoring by put classes in correct modules
Splitted the original MathMLTools classes to mathml-core and mathml-utils.

AndreG-P committed:
Add gold tools to the project (mathml-gold)

AndreG-P committed:
Renaming and solve inner version conflicts

AndreG-P committed:
Fix checkstyle error

AndreG-P committed:
Refactor all loadings and readers to core
Decided to move all loading, readers and writers for XML/MML to the core module.

AndreG-P committed:
Added jackson.xml dependencies to mathml-gold

AndreG-P committed:
Update versions

AndreG-P committed:
Couple of updates to work with mathosphere/pomlp
Also updated the README.md and fix some bugs.
